src/lib/server/orders.ts

```diff
@@ -1110,6 +1115,12 @@ export async function addOrderPayment(
 				}
 			}
 		},
+		...(paymentMethod === 'free' && {
+			invoice: {
+				number: invoiceNumber,
+				createdAt: new Date()
+			}
+		}),
```

Comment:
Hmm what you can do instead (to avoid too much custom code) is:

- call `addOrderPayment` like normal
- if payment method is free, call `onOrderPayment` immediately after

That way you can keep the order pending initially (and remove the check), you can remove this custom code, you can remove the code that marks the order as paid initially, you can remove the code that marks the payment as already paid, and you can even remove the code that calls `updateAfterOrderPaid` (since it will be called inside `onOrderPayment` already)

Reply:
Tried this

``` 
                if (paymentMethod) {
			    const expiresAt = paymentMethodExpiration(paymentMethod);
			    await addOrderPayment(
				    order,
				    paymentMethod,
				    { currency: 'SAT', amount: partialSatoshis },
				    { session, expiresAt }
			    );
		    }
		    
		    if (paymentMethod === 'free') {
			    await onOrderPayment(order, order.payments[0], {
				    amount: 0,
				    currency: runtimeConfig.mainCurrency
			    });
		    }
```
 but got this error.

```
handleError Error: Sync broken between order and payment
    at onOrderPayment (eval at instantiateModule (file:///D:/PVH%20labs/B2BitcoinBootik/node_modules/.pnpm/vite@4.5.0_@types+node@20.9.0/node_modules/vite/dist/node/chunks/dep-bb8a8339.js:56049:28), <anonymous>:78:11)
```

Reply:
ah yes you would need to add an optional `session` parameter to `onOrderPayment`

If provided, it would use `session`, otherwise, it would use the session in `withTransaction`